### PR TITLE
Tests: tag `test_save_load_fast_init_from_base` as flaky

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

As discussed [internally on slack](https://huggingface.slack.com/archives/C01NE71C4F7/p1707407250079089) with @ydshieh -- this test is known to be flaky for a while.

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for adding! 

A corresponding issue should be created to track this

**Review by ydshieh — APPROVED** [review]
I always request to add `description=xxx` when `is_flaky` is use, no matter what the text is put I will accept :-)

**Comment by amyeroberts** [comment]
>  no matter what the text is put I will accept :-)

@ydshieh 

```
@is_flaky(description="is flaky")
```

;)

**Comment by ydshieh** [comment]
> > no matter what the text is put I will accept :-)
> 
> @ydshieh
> 
> ```
> @is_flaky(description="is flaky")
> ```
> 
> ;)

You are a debug master now @amyeroberts !

File: tests/test_modeling_common.py [hunk 1]
```diff
@@ -381,6 +382,7 @@ def test_gradient_checkpointing_enable_disable(self):
                         m.gradient_checkpointing, f"Module {n} does not have gradient_checkpointing set to False"
                     )
 
+    @is_flaky()
```
Comment (gante) [inline tests/test_modeling_common.py:385]:
```suggestion
    @is_flaky(description="low likelihood of failure, reason not yet discovered"))
```

File: tests/test_modeling_common.py [hunk 2]
```diff
@@ -381,6 +382,7 @@ def test_gradient_checkpointing_enable_disable(self):
                         m.gradient_checkpointing, f"Module {n} does not have gradient_checkpointing set to False"
                     )
 
+    @is_flaky(description="low likelihood of failure, reason not yet discovered"))
```
Comment (gante) [inline tests/test_modeling_common.py:385]:
```suggestion
    @is_flaky(description="low likelihood of failure, reason not yet discovered")
```

**Pull request merged**